protocols/interfaces.go

```diff
@@ -88,7 +88,7 @@ type Objective interface {
 
 	Approve() Objective                                                  // returns an updated Objective (a copy, no mutation allowed), does not declare effects
 	Reject() (Objective, SideEffects)                                    // returns an updated Objective (a copy, no mutation allowed), does not declare effects
-	Update(event ObjectiveEvent) (Objective, error)                      // returns an updated Objective (a copy, no mutation allowed), does not declare effects
+	Update(payload ObjectivePayload) (Objective, error)                  // returns an updated Objective (a copy, no mutation allowed), does not declare effects
```

Comment:
Maybe we should rename this to `UpdateWithPayload`

Reply:
I think it is OK as is.